enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,189 @@
+
+### Configuration
+
+The OCM API endpoint will be configurable in the Insights operator via the `support` secret (this is curenlty the only way to configure the Insights operator).
+
+### Disconnected clusters
+
+This feature will not work on disconnected clusters because there is no corresponding token (in the `pull-secret`) to communicate with Red Hat. We should emphasize this in https://docs.openshift.com/container-platform/4.9/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html.
+
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify that the Insights Operator is able to download and update the `pull-secret` content
+
+### Graduation Criteria
+
+Thera are no graduation criteria. This feature is planned as a fast track feature.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+Removing this feature will require updating the Insights operator code base.
+
+### Upgrade / Downgrade Strategy
+
+There is no upgrade/downgrade strategy needed.
+
+### Version Skew Strategy
+
+The Inisghts operator expects the `pull-secret` in the `openshift-config` namespace. If there are any changes to the `pull-secret` location or content, then the update of the Insights operator will be required.
+
+The OCM API is versioned and is expected to be backward compatible. A major update of the OCM API will probably require an update of the Insights operator.
+
+### Operational Aspects of API Extensions
+
+No API extension.
+
+#### Failure Modes
+
+There is no plan to provide direct confirmation of a successful `pull-secret` update. The problem is that the Insights operator is unable to provide such confirmation, because the update is followed up by the MachineConfig operator work. Some potential failures/problems are:  
+
+- There are some paused `MachineConfigPools`, which failed to pick up the updated `pull-secret` definition. The old original secret will
+still be valid for some period of time, but this can happen regardless of the period.
+
+Open question: What will happen in this case? What can we do to mitigate?
+
+- Attacker who can forge an X.509 cert and impersonate OCM (or who gains server-side access to OCM) recommends an invalid pull secret.
+
+#### Support Procedures
+
+A new Insights rule can be created based on the alerts described in the [Monitoring](#monitoring) chapter.
+
+## Implementation History
+
+There are no major milestones in the implementation history.
+
+## Drawbacks
+
+The performance of the OCM API can be a possible drawback.
+
+## Alternatives
+
+- In the future, we may have more flow of data from the OCM API to a cluster and we will revisit if we need a dedicated operator for such flows. However, we do not want to add an entirely new operator for the `pull-secret` flow today.
```

Comment:
While poking around in GitHub today, I stumbled across [ocm-agent][1].  Possibly worth talking to those folks about scoping and timing to see if you want to revisit this alternative or not.

[1]: https://github.com/openshift/ocm-agent

Reply:
Thanks for pointing this out. The OCM folks are reviewing this enhancement, so they can comment. cc: @markturansky

Reply:
sounds good. future requirements TBD.